internal/profiles/service.go

```diff
@@ -126,7 +126,10 @@ func (p *profileService) CreateProfile(
 	}
 
 	// Adds default rule names, if not present
-	PopulateRuleNames(profile)
+	PopulateRuleNames(profile, rulesInProf)
```

Comment:
GIven that `PopulateRuleNames` already uses `profile` as an in-out parameter, would it make sense to do the same for `rulesInProf` (i.e. update the map in `PopulateRuleNames` -- it looks like we're always calling `getUpdatedRuleMapping` and replacing the name afterwards; it feels like we could skip a second for loop by doing so)